test/data/generate_seqr_project_data.py

```diff
@@ -0,0 +1,400 @@
+    sequencing_platforms = enum_resp['enum']['sequencingPlatform']
+    sequencing_types = ['genome', 'exome', 'transcriptome']
+    assay_type = 'sequencing'
+
+    def generate_sequencing_type():
+        """Return a random length of random sequencing types"""
+        k = random.choices(
+            list(sequencing_group_count_distribution.keys()),
+            list(sequencing_group_count_distribution.values()),
+        )[0]
+        return random.choices(sequencing_types, weights=[0.49, 0.49, 0.02], k=k)
+
+    def generate_random_number_within_distribution():
+        """Return a random number within a distribution"""
+        return random.choices(
+            list(sample_count_distribution.keys()), list(sample_count_distribution.values())
+        )[0]
+
+    existing_projects = await papi.get_my_projects_async()
+    for project in PROJECTS:
+        if project not in existing_projects:
+            await papi.create_project_async(
+                name=project, dataset=project, create_test_project=False
+            )
+            logging.info(f'Created project "{project}"')
+            await papi.update_project_async(
+                project=project, body={'meta': {'is_seqr': 'true'}},
+            )
+            logging.info(f'Set {project} as seqr project')
+
+        project_pedigree = generate_pedigree_rows(num_families=random.randint(1, 10))
+        participant_eids = [row.individual_id for row in project_pedigree]
+
+        pedfile = tempfile.NamedTemporaryFile(mode='w')
+        ped_writer = csv.writer(pedfile, delimiter='\t')
+        for row in project_pedigree:
+            ped_writer.writerow(row)
+        pedfile.flush()
+
+        with open(pedfile.name) as f:
+            await FamilyApi().import_pedigree_async(
+                project=project, file=f, has_header=False, create_missing_participants=True
+            )
+
+        id_map = await ParticipantApi().get_participant_id_map_by_external_ids_async(
+            project=project, request_body=participant_eids
+        )
+
+        samples = []
+        for participant_eid in participant_eids:
+            pid = id_map[participant_eid]
+
+            nsamples = generate_random_number_within_distribution()
+            for i in range(nsamples):
+                sample = SampleUpsert(
+                    external_id=f'{participant_eid}_{i+1}',
+                    type=random.choice(sample_types),
+                    meta={
+                        'collection_date': datetime.datetime.now()
+                        - datetime.timedelta(minutes=random.randint(-100, 10000)),
+                        'specimen': random.choice(
+                            ['blood', 'phlegm', 'yellow bile', 'black bile']
+                        ),
+                    },
+                    participant_id=pid,
+                    sequencing_groups=[],
+                )
+                samples.append(sample)
+
+                for stype in generate_sequencing_type():
```

Comment:
The main function seems a bit too long. 
I would suggest to wrap those 2 most inner loops in a function (for stype: for _ ). Will be easier to follow.
Maybe even the whole samples list creation can be refactored as a function.

Otherwise looking good to me.

Reply:
Thanks for the feedback Milo. Agreed that the main function was too long, I've moved the `samples` list creation into its own function and added some better comments which hopefully make it easier to follow.

Reply:
I'd maybe clean it up a bit more. Break down the main function so it just makes function calls. Each commented block should be its own function. It just makes things so much easier to interpret.

I suspect this will be used (and potentially modified) a lot, so it's worth setting is up well if people come back and want to make significant changes later on.